user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/registration/MeasurementUpload.java

```diff
@@ -0,0 +1,424 @@
+    var fileName = succeededEvent.getFileName();
+    Optional<AcceptedFileType> knownFileType = AcceptedFileType.forMimeType(
+        succeededEvent.getMIMEType());
+    if (knownFileType.isEmpty()) {
+      displayError("Unsupported file type. Please make sure to upload a TSV or XLSX file.");
+      return;
+    }
+
+    var parsingResult = switch (knownFileType.get()) {
+      case EXCEL -> parseXLSX(uploadBuffer.inputStream(fileName).orElseThrow());
+      case TSV -> parseTSV(uploadBuffer.inputStream(fileName).orElseThrow());
+    };
+
+    var result = converter.convert(parsingResult);
+
+    var itemsToValidate = result.size();
+
+    var requests = new ArrayList<ValidationRequest>();
+    for (var registration : result) {
+      var request = new ValidationRequest(context.projectId().orElseThrow().value(),
+          context.experimentId().orElseThrow().value(), registration);
+      requests.add(request);
+    }
+    runValidation(requests, itemsToValidate, fileName);
+  }
+
+  private void runValidation(ArrayList<ValidationRequest> requests, int itemsToValidate,
+      String fileName) {
+    AtomicInteger counter = new AtomicInteger();
+    counter.set(1);
+
+    List<ValidationResponse> responses = new ArrayList<>();
+    service.validate(Flux.fromIterable(requests))
+        .doFirst(() -> {
+          showValidationProgress();
+          setValidationProgressText("Starting validation ...");
+        })
+        .doOnNext(item -> setValidationProgressText(
+            "Processed " + counter.getAndIncrement() + " requests from " + itemsToValidate))
+        .doOnNext(responses::add)
+        .doOnComplete(() -> {
+          hideValidationProgress();
+          displayValidationResults(responses, itemsToValidate, fileName);
+        })
+        .subscribe();
+  }
+
+  private void hideValidationProgress() {
+    getUI().ifPresent(ui -> ui.access(() -> validationProgress.setVisible(false)));
+  }
+
+  private void showValidationProgress() {
+    getUI().ifPresent(ui -> ui.access(() -> validationProgress.setVisible(true)));
+  }
+
+  private void setValidationProgressText(String text) {
+    getUI().ifPresent(ui -> ui.access(() -> validationProgress.setText(text)));
+  }
+
+  private void displayValidationResults(List<ValidationResponse> responses, int totalValidations,
+      String fileName) {
+    ValidationResult result = ValidationResult.successful();
+    var combinedResult = responses.stream().map(ValidationResponse::result)
+        .reduce(result, ValidationResult::combine);
+
+    getUI().ifPresent(ui -> ui.access(() -> {
+      var measurementFileItem = new MeasurementFileItem(fileName,
+          new MeasurementValidationReport(totalValidations, combinedResult));
+      addAndDisplayFile(measurementFileItem);
+    }));
+  }
+
+  private void addAndDisplayFile(MeasurementFileItem measurementFileItem) {
+    measurementFileItems.add(measurementFileItem);
+    showFile(measurementFileItem);
```

Comment:
why did you decide to separate this into two methods?

Reply:
legacy code from the upload dialog. But i see that it does not make sense to offer the showFile method explicitly